Added env var config to plugin manager

Nabsku opened the pull request:
## Summary

This PR adds the ability to configure the plugin manager via env vars. This allows the user to set a proxy or even set npm settings like the registry. 

## Changes

- Added templating for env vars in the plugin manager

## Steps to Test

1. Set .Values.pluginsManager.env
2. Watch the envs vars get populated

Nabsku commented:
> ## Pull Request Overview
> Adds support for configuring the plugin manager via environment variables by updating Helm chart defaults and templates.
> 
> * Introduces a commented-out `env` block in `values.yaml` for users to specify plugin manager env vars.
> * Updates the deployment template to render `.Values.pluginsManager.env` under the container spec.
> 
> ### Reviewed Changes
> Copilot reviewed 2 out of 2 changed files in this pull request and generated no comments.
> 
> File	Description
> charts/headlamp/values.yaml	Added instructions for setting `pluginsManager.env`
> charts/headlamp/templates/deployment.yaml	Injects `env:` entries from `.Values.pluginsManager.env`
> Comments suppressed due to low confidence (2)

Just to reference the two low confidence comments: I decided not to do that and follow the existing conventions in the pluginManager (see `resources:`). If wanted, I can of course add conditionals everywhere + apply that to the whole chart.

illume commented:
@Nabsku Thanks for this.

Would you mind adding that variable to the README.md in the charts/headlamp folder?
https://github.com/kubernetes-sigs/headlamp/blob/main/charts/headlamp/README.md

Nabsku commented:
@illume @skoeva sorry for the long delay! I cleaned up the commits, signed them and added the requested documentation section. 

Let me know if I missed something!
